src/components/AddBook/AddBook.js

```diff
@@ -1,18 +1,66 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addBook } from '../../redux/books/books';
 import style from './AddBook.module.css';
 
-const AddBook = () => (
-  <>
-    <div className={style.dividerHorizontal} />
-    <section className={style.Container}>
-      <h2 className={style.formTitle}>ADD NEW BOOK</h2>
-      <form action="./" className={style.addBookForm}>
-        <input className={`${style.input} ${style.titleInput}`} type="text" placeholder="Book title" required Value />
-        <input className={`${style.input} ${style.titleInput}`} type="text" placeholder="Book Author" required Value />
-        <button type="button" className={style.btnAddBook}>ADD BOOK</button>
-      </form>
-    </section>
-  </>
-);
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const bookState = useSelector((state) => state.books);
+  const [book, setBook] = useState({
+    id: bookState.length + 1,
+    title: '',
+    author: '',
+  });
+  const handleChangeinput = (e) => {
+    const fieldName = e.target.name;
+    if (fieldName === 'title') {
+      setBook({ title: e.target.value, author: book.author });
+    }
+    if (fieldName === 'author') {
+      setBook({ title: book.title, author: e.target.value });
+    }
+  };
+
+  const handleSubmitbtn = (e) => {
+    e.preventDefault();
+    dispatch(addBook(book));
+  };
+  return (
+    <>
+      <div className={style.dividerHorizontal} />
+      <section className={style.Container}>
+        <h2 className={style.formTitle}>ADD NEW BOOK</h2>
+        <form action="./" className={style.addBookForm}>
+          <input
+            className={`${style.input} ${style.titleInput}`}
+            name="title"
+            id="bookTitle"
+            type="text"
+            placeholder="Book title"
+            required
+            value
+            onChange={handleChangeinput}
+          />
+          <input
+            className={`${style.input} ${style.titleInput}`}
+            name="author"
+            type="text"
+            placeholder="Book Author"
+            required
+            value
+            onChange={handleChangeinput}
+          />
+          <button
+            type="button"
+            className={style.btnAddBook}
+            onClick={handleSubmitbtn}
+          >
+            ADD BOOK
+          </button>
+        </form>
```

Comment:
- [ ] I'm getting values true on the add book form and I'm not able to edit these values. Please fix this issue and make sure the field values are empty and users are able to add book data into fields.
 - I would suggest using two-state for these fields and on the input `value` attribute place the state value and change the state on the change event.

![image](https://user-images.githubusercontent.com/83803414/178687429-eb3a2872-5379-44e1-97dc-dd8a0704c34b.png)

Reply:
@caasperr Hi, 
`const [title, setTitle] = useState('')`

` value={title} onChange={(e) => setTitle(e.target.value)}`

You can implement the same for the author.

Reply:
@thecodechaser, Thank you so much for elaborating. but I still have a question why use 2 states when we can handle both with objects in 1line? why add 3-4 extra?

Reply:
@caasperr That's only a suggestion 😄